earth_enterprise/src/common/config/geConfigUtil.cpp

```diff
@@ -41,41 +41,14 @@ bool IsRedHat(void) {
 }
 
 uint GetNumCPUs(void) {
-  uint numcpus = 0;
-  std::ifstream in("/proc/cpuinfo");
-  if (in) {
-    std::string line;
-    while (std::getline(in, line)) {
-      if (StartsWith(line, "processor")) {
-        ++numcpus;
-      }
-    }
-  }
-
-  return std::max(uint(1), numcpus);
+  return std::max(uint(1), uint(sysconf(_SC_NPROCESSORS_ONLN)));
```

Comment:
I'm a bit concerned about this since `_SC_NPROCESSORS_ONLN` is a non-standard GNU extension. Where possible I'd prefer to write code that works on all systems. As messy as it is, that might be the reason for the original code that reads through /proc.

Same comment applies to the other `sysconf` calls.

Reply:
Understood. There is also pulling in the results from a call to `nproc`. `std::thread::hardware_concurrency()` is another option, but it would necessitate waiting on c++11 migration to be complete